pyproject.toml

```diff
@@ -47,6 +47,45 @@ repository = "https://github.com/fox-it/dissect.target"
 [project.optional-dependencies]
 full = [
     "asn1crypto",
+    "dissect.btrfs>=1,<2",
+    "dissect.cim>=3,<4",
+    "dissect.clfs>=1,<2",
+    "dissect.esedb>=3,<4",
+    "dissect.etl>=3,<4",
+    "dissect.extfs>=3,<4",
+    "dissect.fat>=3,<4",
+    "dissect.ffs>=3,<4",
+    "dissect.jffs>=1,<2",
+    "dissect.shellitem>=3,<4",
+    "dissect.squashfs>=1,<2",
+    "dissect.sql>=3,<4",
+    "dissect.thumbcache>=1,<2",
+    "dissect.vmfs>=3,<4",
+    "dissect.xfs>=3,<4",
+    "ipython",
+    "fusepy",
+    "pycryptodome",
+    "ruamel.yaml",
+    "tomli; python_version<'3.11'",
+    # dissect.target's caching uses flow.record functionlity which depends on the
+    # zstandard module being available. However flow.record does not define
+    # zstandard as a dependency, nor does it allow zstandard to be installed
+    # through extras.
+    #
+    # Until such time that this dependency can be installed through
+    # flow.record, we define it as a dependency of dissect.target.
+    "zstandard",
+]
+dev = [
```

Comment:
They should all (except target, cstruct and util) specify the `[dev]` extra to make sure the initial `4.0.dev` versions of `dissect.cstruct` can be selected.